### PR TITLE
check error when parse field failed

### PR DESCRIPTION
**Pull request by hanxiaoshuai**
**What this PR does / why we need it**:
check error when parse field failed
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by hanxiaoshuai** [comment]
cc @CaoShuFeng I rebased it. 
/assign @deads2k @smarterclayton

File: pkg/kubectl/sorter.go [hunk 1]
```diff
@@ -277,13 +277,16 @@ func (r *RuntimeSort) Less(i, j int) bool {
...
-	parser.Parse(r.field)
-
 	var iValues [][]reflect.Value
 	var jValues [][]reflect.Value
 	var err error
 
+	parser := jsonpath.New("sorting").AllowMissingKeys(true)
+	err = parser.Parse(r.field)
+	if err != nil {
+		glog.Fatalf("Failed to parse field:%s err: %v", r.field, err)
```
Comment (deads2k) [inline pkg/kubectl/sorter.go]: In general, I prefer a panic.

If you try to allow missing keys, why wouldn't you allow a parsing failure?

**Comment by deads2k** [comment]
needs a squash and a review for intent

/unassign
/assign @soltysh 
/assign @juanvallejo

**Review by juanvallejo — APPROVED** [review]

**Comment by soltysh** [comment]
/hold 
This requires previous comments addressed and updates so it's green.

File: pkg/kubectl/sorter.go (hunk 1 above)
Comment (hanxiaoshuai) [reply pkg/kubectl/sorter.go]: Done

**Review by soltysh — APPROVED** [review]
/hold cancel
/lgtm

**Pull request merged**